pkg/nar/header.go

```diff
@@ -15,6 +17,44 @@ type Header struct {
 	Executable bool     // Set to true for files that are executable
 }
 
+// Validate does some consistency checking of the header structure, such as
+// checking for valid paths and inconsistent fields, and returns an error if it
+// fails validation.
+func (h *Header) Validate() error {
+	// Path may not start with a /, and may not contain null bytes
+	if len(h.Path) > 1 {
+		if h.Path[:1] == "/" {
+			return fmt.Errorf("path %v starts with a /", h.Path)
+		}
+
+		if strings.ContainsAny(h.Path, "\u0000") {
+			return fmt.Errorf("path contains null bytes")
+		}
+	}
+
+	// Regular files and directories may not have LinkTarget set.
+	if h.Type == TypeRegular || h.Type == TypeDirectory {
+		if h.LinkTarget != "" {
+			if h.LinkTarget != "" {
+				return fmt.Errorf("type is %v, but LinkTarget is not empty", h.Type.String())
+			}
+		}
+	}
+
+	// Directories and Symlinks may not have Size and Executable set.
+	if h.Type == TypeDirectory || h.Type == TypeSymlink {
+		if h.Size != 0 {
+			return fmt.Errorf("type is %v, but Size is not 0", h.Type.String())
+		}
+
+		if h.Executable {
+			return fmt.Errorf("type is %v, but Executable is true", h.Type.String())
+		}
+	}
+
```

Comment:
Currently empty link targets are accepts for `TypeSymlink`

Reply:
nice catch! fixed.